src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCacheableResponse.java

```diff
@@ -0,0 +1,720 @@
+    private int padding;
+
+    private final int initialWindow;
+    private int payloadWritten = -1;
+    private DefaultCacheEntry cacheEntry;
+    private boolean etagSent;
+    private long traceId;
+    private boolean isResponseBuffering;
+
+    HttpCacheProxyCacheableResponse(
+        HttpCacheProxyFactory httpCacheProxyFactory,
+        HttpCacheProxyCacheableRequest request,
+        MessageConsumer connectReplyThrottle,
+        long connectRouteId,
+        long connectReplyId,
+        long acceptInitialId)
+    {
+        this.streamFactory = httpCacheProxyFactory;
+        this.request = request;
+        this.connectReplyThrottle = connectReplyThrottle;
+        this.connectRouteId = connectRouteId;
+        this.connectReplyStreamId = connectReplyId;
+        this.acceptInitialId = acceptInitialId;
+        this.streamState = this::beforeBegin;
+
+        this.initialWindow = this.streamFactory.responseBufferPool.slotCapacity();
+    }
+
+    void handleStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        streamState.accept(msgTypeId, buffer, index, length);
+    }
+
+    @Override
+    public String toString()
+    {
+        return String.format("%s[connectRouteId=%016x, connectReplyStreamId=%d, acceptReplyBudget=%016x, " +
+                "connectReplyBudget=%d, padding=%d]", getClass().getSimpleName(),
+            connectRouteId, connectReplyStreamId, acceptReplyBudget, connectReplyBudget, padding);
+    }
+
+    private void beforeBegin(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        if (msgTypeId == BeginFW.TYPE_ID)
+        {
+            final BeginFW begin = this.streamFactory.beginRO.wrap(buffer, index, index + length);
+            handleBegin(begin);
+        }
+        else
+        {
+            this.streamFactory.writer.doReset(connectReplyThrottle, connectRouteId, connectReplyStreamId,
+                    streamFactory.supplyTrace.getAsLong());
+        }
+    }
+
+    private void handleBegin(
+        BeginFW begin)
+    {
+        final long connectReplyId = begin.streamId();
+        traceId = begin.trace();
+
+        final OctetsFW extension = streamFactory.beginRO.extension();
+
+        if (extension.sizeof() > 0)
+        {
+            final HttpBeginExFW httpBeginFW = extension.get(streamFactory.httpBeginExRO::wrap);
+
+            if (DEBUG)
+            {
+                System.out.printf("[%016x] CONNECT %016x %s [received response]\n", currentTimeMillis(), connectReplyId,
+                        getHeader(httpBeginFW.headers(), ":status"));
+            }
+
+            final ListFW<HttpHeaderFW> responseHeaders = httpBeginFW.headers();
+            handleInitialRequest(responseHeaders);
+        }
+        else
+        {
+            this.streamFactory.writer.doReset(connectReplyThrottle, connectRouteId, connectReplyStreamId, traceId);
+        }
```

Comment:
When this code is moved to stream creation, no need to explicitly send `RESET` because that is handled automatically by `reaktor` if returned stream is `null`.